### PR TITLE
VEOSS-1816 Alter Dashboard Variations for Excluded Traders

### PR DESCRIPTION
**Pull request by ZDevelop94**
- Needs Removal of optional `effectiveDate` in `ExcludedTrader` in registration backend to be merged.

### CODE REVIEW

File: app/views/IndexView.scala.html [hunk 1]
```diff
@@ -236,8 +249,29 @@ <h2 class="govuk-heading-m">@messages("index.details")</h2>
                 </p>
             }
 
-            @(maybeExcludedTrader, exclusionsEnabled) match {
-                case (Some(_), true) => {}
+            @(maybeExcludedTrader, exclusionsEnabled, hasRequestedToLeave) match {
+                case (Some(excludedTrader), true, false) => {
+                @if(hasSubmittedFinalReturn){
+                    @if(excludedTrader.exclusionReason == 4) {
```
Comment (DevAndyroo) [inline app/views/IndexView.scala.html:255]: if we could use the enum here as well, might be better. I'd be open for you to change the backend to provide it as an enum as well, over an int

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/services/PartialReturnPeriodService.scala [hunk 2]
```diff
@@ -36,14 +36,16 @@ class PartialReturnPeriodService @Inject()(
       case None =>
         getMaybeFirstPartialReturnPeriod(registration)
       case Some(excludedTrader) =>
+        val hmrcExclusionReason = 6
```
[the remark this answers is not in the document]
Comment (ZDevelop94) [reply app/services/PartialReturnPeriodService.scala:39]: Yeah I created enums i will use them

File: app/views/IndexView.scala.html [hunk 3]
```diff
@@ -155,7 +172,7 @@ <h2 class="govuk-heading-m">@messages("index.yourReturns")</h2>
                 @if(exclusionsEnabled) {
                     @maybeExcludedTrader match {
                         case Some(excludedTrader) if currentReturnIsFinal => {
-                            <p class="govuk-body">@messages("yourAccount.finalReturn.message")</p>
+                            <p class="govuk-body">@messages("yourAccount.finalReturn.message")</p> //here
```
Comment (DevAndyroo) [inline app/views/IndexView.scala.html:175]: one last one

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/views/IndexView.scala.html (hunk 3 above)
Comment (ZDevelop94) [reply app/views/IndexView.scala.html:175]: Weird not on my one refresh?

**Pull request merged**